src/docker/shared.rs

```diff
@@ -28,6 +28,217 @@ const DOCKER_IMAGES: &[&str] = &include!(concat!(env!("OUT_DIR"), "/docker-image
 // to fork the process, and which podman allows by default.
 pub(crate) const SECCOMP: &str = include_str!("seccomp.json");
 
+#[derive(Debug)]
+pub struct DockerOptions {
+    engine: Engine,
+    target: Target,
+    config: Config,
+    uses_xargo: bool,
+}
+
+impl DockerOptions {
+    pub fn new(engine: Engine, target: Target, config: Config, uses_xargo: bool) -> DockerOptions {
+        DockerOptions {
+            engine,
+            target,
+            config,
+            uses_xargo,
+        }
+    }
+
+    pub fn in_docker(&self) -> bool {
```

Comment:
These getters should probably be removed since we always pass by a non-mutable reference and these structs are only used internally, so it's entirely redundant.